Ryujinx/Ui/NpadController.cs

```diff
@@ -1,160 +1,62 @@
-        public ControllerInputId StickButton;
-        public ControllerInputId ButtonMinus;
-        public ControllerInputId ButtonL;
-        public ControllerInputId ButtonZl;
-        public ControllerInputId DPadUp;
-        public ControllerInputId DPadDown;
-        public ControllerInputId DPadLeft;
-        public ControllerInputId DPadRight;
-    }
+        private InnerNpadController _inner;
 
-    public struct NpadControllerRight
-    {
-        public ControllerInputId Stick;
-        public ControllerInputId StickButton;
-        public ControllerInputId ButtonA;
-        public ControllerInputId ButtonB;
-        public ControllerInputId ButtonX;
-        public ControllerInputId ButtonY;
-        public ControllerInputId ButtonPlus;
-        public ControllerInputId ButtonR;
-        public ControllerInputId ButtonZr;
-    }
+        // NOTE: This should be initialized AFTER GTK for compat reasons with OpenTK SDL2 backend and GTK on Linux.
+        // BODY: Usage of Joystick.GetState must be defer to after GTK full initialization. Otherwise, GTK will segfault because SDL2 was already init *sighs*
+        public NpadController(InnerNpadController inner)
+        {
+            _inner = inner;
+        }
 
-    public class NpadController
-    {
-        /// <summary>
-        /// Enables or disables controller support
-        /// </summary>
-        public bool Enabled { get; private set; }
-
-        /// <summary>
-        /// Controller Device Index
-        /// </summary>
-        public int Index { get; private set; }
-
-        /// <summary>
-        /// Controller Analog Stick Deadzone
-        /// </summary>
-        public float Deadzone { get; private set; }
-
-        /// <summary>
-        /// Controller Trigger Threshold
-        /// </summary>
-        public float TriggerThreshold { get; private set; }
-
-        /// <summary>
-        /// Left JoyCon Controller Bindings
-        /// </summary>
-        public NpadControllerLeft LeftJoycon { get; private set; }
-
-        /// <summary>
-        /// Right JoyCon Controller Bindings
-        /// </summary>
-        public NpadControllerRight RightJoycon { get; private set; }
-
-        public NpadController(
-            bool                enabled,
-            int                 index,
-            float               deadzone,
-            float               triggerThreshold,
-            NpadControllerLeft  leftJoycon,
-            NpadControllerRight rightJoycon)
+        private bool IsEnabled()
         {
-            Enabled          = enabled;
-            Index            = index;
-            Deadzone         = deadzone;
-            TriggerThreshold = triggerThreshold;
-            LeftJoycon       = leftJoycon;
-            RightJoycon      = rightJoycon;
+            return _inner.Enabled && Joystick.GetState(_inner.Index).IsConnected;
         }
 
-        public void SetEnabled(bool enabled)
+        public void Initialize()
```

Comment:
```suggestion
        public void Initialize() { }
```
But maybe it’s fine to remove it, since it doesn’t do anything!